python/kserve/kserve/protocol/rest/openai/openai_proxy_model.py

```diff
@@ -254,16 +254,20 @@ async def create_completion(
             )
             return it
         else:
-            response = await self._http_client.send(req)
-            response.raise_for_status()
-            if self.skip_upstream_validation:
-                obj = response.json()
-                completion = Completion.model_construct(**obj)
-            else:
-                completion = Completion.model_validate_json(response.content)
+            completion = await self.predict_completion(req)
             self.postprocess_completion(completion, request)
             return completion
 
+    async def predict_completion(self, req: httpx.Request) -> Completion:
```

Comment:
not sure if predict is the right verb, should we name it `generate_completion` ?

Reply:
Also we should't expose http client to the function signature